Tools/autotest/arducopter.py

```diff
@@ -1934,6 +1934,194 @@ def autotest(self):
             return False
         return True
 
+    def arm_motors_with_rc_input_heli(self):
+        """Arm motors with radio."""
+        self.progress("Arm motors with radio")
+        self.set_rc(4, 2000)
+        tstart = self.get_sim_time()
+        timeout = 15
+        while self.get_sim_time() < tstart + timeout:
+            self.mav.wait_heartbeat()
+            if self.mav.motors_armed():
+                self.progress("MOTORS ARMED OK WITH RADIO")
+                self.set_rc(4, 1500)
+                return True
+        self.progress("FAILED TO ARM WITH RADIO")
+        self.set_rc(4, 1500)
+        return False
+
+    def disarm_motors_with_rc_input_heli(self):
+        """Disarm motors with radio."""
+        self.progress("Disarm motors with radio")
+        self.set_rc(4, 1000)
+        tstart = self.get_sim_time()
+        timeout = 15
+        while self.get_sim_time() < tstart + timeout:
+            self.mav.wait_heartbeat()
+            if not self.mav.motors_armed():
+                self.progress("MOTORS DISARMED OK WITH RADIO")
+                self.set_rc(4, 1500)
+                return True
+        self.progress("FAILED TO DISARM WITH RADIO")
+        self.set_rc(4, 1500)
+        return False
+
+    def arm_motors_with_switch_on_CH7(self):
+        """Arm motors with switch"""
+        self.progress("Arm motors with switch")
+        self.set_rc(7, 2000)
+        tstart = self.get_sim_time()
+        timeout = 15
+        while self.get_sim_time() < tstart + timeout:
+            self.mav.wait_heartbeat()
+            if self.mav.motors_armed():
+                self.progress("MOTORS ARMED OK WITH SWITCH")
+                return True
+        self.progress("FAILED TO ARM WITH SWITCH")
+        return False
+
+    def disarm_motors_with_switch_on_CH7(self):
+        """Disarm motors with switch"""
+        self.progress("Disarm motors with switch")
+        self.set_rc(7, 1000)
+        tstart = self.get_sim_time()
+        timeout = 15
+        while self.get_sim_time() < tstart + timeout:
+            self.mav.wait_heartbeat()
+            if not self.mav.motors_armed():
+                self.progress("MOTORS DISARMED OK WITH SWITCH")
+                return True
+        self.progress("FAILED TO DISARM WITH SWITCH")
+        return False
+
+    def arm_checks_heli(self):
+        # TEST ARMING/DISARM
+        self.set_parameter("DISARM_DELAY", 0)
+        self.set_parameter("ARMING_RUDDER", 2)
+        self.progress("Check to verify stick arming")
+        if not self.arm_motors_with_rc_input_heli():
+            self.progress("Aircraft failed to arm using stick arming")
+            self.arm_vehicle()
```

Comment:
Ok. So I think I understand this.  You are saying that once it raises an exception then it exits the routine.  I did this alot because I thought it kept going even though it raises an exception.  So that makes sense.  I will have to go back and change this.  But its ok to do if the aircraft arms when we didn't want it to because that gets it back to an expected state for remaining tests?